libvast/src/system/index.cpp

```diff
@@ -1030,6 +1135,56 @@ index(index_actor::stateful_pointer<index_state> self,
                       e);
           });
     },
+    [self](atom::evaluate, vast::query query,
+           uint64_t n) -> caf::result<query_response> {
+      if (n != 0)
+        return caf::make_error(
+          ec::unimplemented, fmt::format("requesting a result count other then "
+                                         "0 (all) is not supported yet"));
+      if (!self->state.accept_queries) {
+        VAST_VERBOSE("{} delays query {} because it is still starting up",
+                     *self, query);
+        return caf::skip;
+      }
+      // `current_sender()` doesn't work after creating the response promise, so
+      // we must do it before.
+      auto rp = self->make_response_promise<query_response>();
+      std::vector<uuid> candidates;
+      for (const auto& [_, active_partition] : self->state.active_partitions)
+        candidates.push_back(active_partition.id);
+      for (const auto& [id, _] : self->state.unpersisted)
+        candidates.push_back(id);
+      self
+        ->request(self->state.catalog, caf::infinite, atom::candidates_v, query)
+        .then([=, candidates = std::move(candidates)](
+                catalog_result& midx_result) mutable {
+          auto& midx_candidates = midx_result.partitions;
+          VAST_DEBUG("{} got initial candidates {} and from meta-index {}",
+                     *self, candidates, midx_candidates);
+          candidates.insert(candidates.end(), midx_candidates.begin(),
+                            midx_candidates.end());
+          std::sort(candidates.begin(), candidates.end());
+          candidates.erase(std::unique(candidates.begin(), candidates.end()),
+                           candidates.end());
```

Comment:
Can the meta index return duplicate candidate partitions?

Reply:
Not as far as I know, I simply inherited this part from the old implementation.